lib/api/core/validation/types/geoShape.js

```diff
@@ -1,196 +1,223 @@
-    result = false;
-  }
+    if (shape.type !== 'geometrycollection' && (!shape.coordinates || !Array.isArray(shape.coordinates))) {
+      errorMessages.push(`The coordinates property must be provided for the "${shape.type}" shape type.`);
+      result = false;
+    }
 
-  if (shape.type !== 'geometrycollection' && (!shape.coordinates || !Array.isArray(shape.coordinates))) {
-    errorMessages.push(`The coordinates property must be provided for the "${shape.type}" shape type.`);
-    result = false;
-  }
+    if (shape.type === 'circle' && !shape.radius) {
+      errorMessages.push('The radius property is mandatory for the "circle" shape type.');
+      result = false;
+    }
 
-  if (shape.type === 'circle' && !shape.radius) {
-    errorMessages.push('The radius property is mandatory for the "circle" shape type.');
-    result = false;
-  }
+    if (shape.type !== 'circle' && shape.radius) {
+      errorMessages.push(`The radius property must not be provided for the "${shape.type}" shape type.`);
+      result = false;
+    }
 
-  if (shape.type !== 'circle' && shape.radius) {
-    errorMessages.push(`The radius property must not be provided for the "${shape.type}" shape type.`);
-    result = false;
-  }
+    if (shape.type !== 'polygon' && shape.type !== 'multipolygon' && shape.orientation) {
+      errorMessages.push(`The orientation property must not be provided for the "${shape.type}" shape type.`);
+      result = false;
+    }
 
-  if (shape.type !== 'polygon' && shape.type !== 'multipolygon' && shape.orientation) {
-    errorMessages.push(`The orientation property must not be provided for the "${shape.type}" shape type.`);
-    result = false;
-  }
+    if (shape.type !== 'geometrycollection' && shape.geometries) {
+      errorMessages.push(`The geometries property must not be provided for the "${shape.type}" shape type.`);
+      result = false;
+    }
 
-  if (shape.type !== 'geometrycollection' && shape.geometries) {
-    errorMessages.push(`The geometries property must not be provided for the "${shape.type}" shape type.`);
-    result = false;
-  }
+    if (shape.type === 'geometrycollection' && (!shape.geometries || !Array.isArray(shape.geometries))) {
+      errorMessages.push('The geometries property must be provided for the "geometrycollection" shape type.');
+      result = false;
+    }
 
-  if (shape.type === 'geometrycollection' && (!shape.geometries || !Array.isArray(shape.geometries))) {
-    errorMessages.push('The geometries property must be provided for the "geometrycollection" shape type.');
-    result = false;
+    return result;
   }
 
-  return result;
-};
+  /**
+   * @param {TypeOptions} typeOptions
+   * @return {boolean|TypeOptions}
+   * @throws InternalError
+   */
+  validateFieldSpecification(typeOptions) {
+    if (typeOptions.hasOwnProperty('shapeTypes')) {
+      if (!Array.isArray(typeOptions.shapeTypes) || typeOptions.shapeTypes.length === 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 90 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)